src/core/Akka.Cluster/Cluster.cs

```diff
@@ -128,9 +133,14 @@ public void Subscribe(IActorRef subscriber, Type[] to)
         /// If set to <see cref="ClusterEvent.SubscriptionInitialStateMode.InitialStateAsSnapshot"/> 
         /// a snapshot of <see cref="ClusterEvent.CurrentClusterState"/> will be sent to <paramref name="subscriber"/> as the first message. </param>
         /// <param name="to"><see cref="ClusterEvent.IClusterDomainEvent"/> subclasses</param>
-        public void Subscribe(IActorRef subscriber, ClusterEvent.SubscriptionInitialStateMode initialStateMode, Type[] to)
+        public void Subscribe(IActorRef subscriber, ClusterEvent.SubscriptionInitialStateMode initialStateMode, params Type[] to)
         {
-            ClusterCore.Tell(new InternalClusterAction.Subscribe(subscriber, initialStateMode, ImmutableHashSet.Create<Type>(to)));
+            if (to.Length == 0)
+                throw new ArgumentException("At least one `IClusterDomainEvent` class is required");
+            if (!to.All(c => c.IsAssignableFrom(typeof(ClusterEvent.IClusterDomainEvent))))
```

Comment:
We should the the opposite comparsion
```
!to.All(t => typeof(ClusterEvent.IClusterDomainEvent).IsAssignableFrom(t))
```

Maybe we have the same issues in another places